### ISSUE TITLE
azurerm crash if azurerm_orbital_contact_profile is modified

### ISSUE THREAD

**Issue by rmelick-muon** [hashicorp/terraform-provider-azurerm#24311]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.3.9

### AzureRM Provider Version

3.85.0

### Affected Resource(s)/Data Source(s)

azurerm_orbital_contact_profile

### Terraform Configuration Files

```hcl
See https://github.com/hashicorp/terraform-provider-azurerm/issues/24310 for the resource setup
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.85.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x106935af0]

goroutine 1895 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/orbital.flattenContactProfileChannel({0x14002a0eb00?, 0x2, 0x0?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/orbital/helper.go:290 +0x140
github.com/hashicorp/terraform-provider-azurerm/internal/services/orbital.flattenContactProfileLinks({0x1400585b500?, 0x1, 0x1400020a7e0?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/orbital/helper.go:268 +0x164
github.com/hashicorp/terraform-provider-azurerm/internal/services/orbital.(*ContactProfileResource).Read.ContactProfileResource.Read.func1({0x108e77080, 0x1400020a7e0}, {0x14002923200, {0x108e797a8, 0x1400030e060}, 0x14005811c00, 0x0, {0x108e79968, 0x10e96c780}})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/orbital/contact_profile_resource.go:198 +0x228
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func3({0x108e77080, 0x1400020a7e0}, 0x1400020a7e0?, {0x107f67260?, 0x14002923200?})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:68 +0xf0
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.(*ResourceWrapper).diagnosticsWrapper.diagnosticsWrapper.func11({0x108e77080?, 0x1400020a7e0?}, 0x0?, {0x107f67260?, 0x14002923200?})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:190 +0x4c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x140007cb960, {0x108e76fd8, 0x14003399170}, 0xd?, {0x107f67260, 0x14002923200})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:795 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x140007cb960, {0x108e76fd8, 0x14003399170}, 0x140033692b0, {0x107f67260, 0x14002923200})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:1089 +0x430
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x1400047d848, {0x108e76fd8?, 0x14003399080?}, 0x14005986d00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:649 +0x3e4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x140010e4fa0, {0x108e76fd8?, 0x140033988d0?}, 0x140027e5500)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x390
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x108a762e0?, 0x140010e4fa0}, {0x108e76fd8, 0x140033988d0}, 0x14001455b20, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002a41e0, {0x108e88880, 0x14000e28b60}, 0x1400211b320, 0x140011f95f0, 0x10e926c48, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
google.golang.org/grpc.(*Server).handleStream(0x140002a41e0, {0x108e88880, 0x14000e28b60}, 0x1400211b320, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1751 +0x7c4
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 55
	google.golang.org/grpc@v1.58.3/server.go:997 +0x160

Error: The terraform-provider-azurerm_v3.85.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.


```

### Expected Behaviour

Because of https://github.com/hashicorp/terraform-provider-azurerm/issues/24310, I needed to manually modify the `demodulationConfiguration` field from the Azure portal, to make it blank on the channel where I wanted to use virtualRF and not use the managed modem configuration.

After doing that, I would expect to still be able to run terraform.

### Actual Behaviour

After making that modification, `terraform plan` crashes with the above error.

### Steps to Reproduce

terraform plan

### References

#24310

**Comment by rmelick-muon** [comment]
@jiaweitao001 I just tested with updating the provider to 3.86.0, and confirm that I no longer see this error when manually modifying demodulation configuration through the Azure Portal, it seems to be fixed.

Thank you for fixing it!

**Comment by rcskosir** [comment]
I am marking this as closed with #24315, thank you all!
